tachyon/math/polynomials/univariate/radix2_evaluation_domain.h

```diff
@@ -242,28 +243,29 @@ class Radix2EvaluationDomain : public UnivariateEvaluationDomain<F, MaxDegree> {
     }
   }
 
-  // Computes the first |size_ / 2| roots of unity for the entire domain.
-  // e.g. for the domain [1, g, g², ..., gⁿ⁻¹}], it computes
+  // Computes the first |size| roots of unity for the entire domain.
+  // e.g. for the domain [1, g, g², ..., gⁿ⁻¹}] and |size| = n / 2, it computes
   // [1, g, g², ..., g^{(n / 2) - 1}]
-  constexpr std::vector<F> GetRootsOfUnity(const F& root) const {
+  constexpr std::vector<F> GetRootsOfUnity(size_t size, const F& root) const {
 #if defined(TACHYON_HAS_OPENMP)
     uint32_t log_size = SafeLog2Ceiling(this->size_);
     if (log_size <= kMinLogRootsOfUnitySizeForParallelization)
 #endif
-      return ComputePowersSerial(this->size_ / 2, root);
+      return ComputePowersSerial(size, root);
 #if defined(TACHYON_HAS_OPENMP)
+    size_t required_log_size = size_t{SafeLog2Ceiling(size)};
     F power = root;
-    // [g, g², g⁴, g⁸, ..., g^(2^(|log_size| - 1))]
+    // [g, g², g⁴, g⁸, ..., g^(2^(|required_log_size|))]
     std::vector<F> log_powers =
-        base::CreateVector(size_t{log_size} - 1, [&power]() {
+        base::CreateVector(required_log_size, [&power]() {
           F old_value = power;
           power.SquareInPlace();
           return old_value;
         });
 
     // allocate the return array and start the recursion
     std::vector<F> powers =
-        base::CreateVector(size_t{1} << (log_size - 1), F::Zero());
+        base::CreateVector(size_t{1} << required_log_size, F::Zero());
```

Comment:
```suggestion
        base::CreateVector(1 << required_log_size, F::Zero());
```

Reply:
This needs to be correctly casted, otherwise it might get overflowed.